### ISSUE TITLE
Opt out a property for equals/hashCode

### ISSUE THREAD

**Issue by fluentfuture** [inferred/FreeBuilder#54]
I would like the ability to define a value class where a certain field isn't included in equality.

For example, I recently came to a need to define a helper class that wraps the input object T and its original index in the input List.

```
IndexedFoo {
  int index();
  Foo foo();
}

Multimap<IndexedFoo, Foo> overlappings = ...;
```

I want IndexedFoo's equality to be based on the index, but not the Foo object.

**Comment by chrisalice** [comment]
Just a note: dropping fields from equality makes it very easy to write broken unit tests (and hard to write good ones). If you find yourself wanting this feature, consider first whether you actually want a different collection type (typically a Map rather than a Set, for instance), or if you can use an explicit field-ignoring Comparator in the parts of the code that need it.

Otherwise, :+1: this issue, and consider adding a comment explaining your situation!

**Comment by thespags** [comment]
@alicederyn sounds like you aren't a fan of the premise but are okay with the feature. In other words, someone could provide a PR for this feature that could be accepted.

For a situation I find myself in:
I have a use case for a lazy loaded graph where the Graph is an attribute of my POJOs for traversing relationships. However two given nodes should be equal even if their Graph attributes aren't in the same state.

**Comment by alicederyn** [comment]
Thank you @thespags for implementing this one, now available in [v2.4.0](https://github.com/inferred/FreeBuilder/releases/tag/v2.4.0)
